py-polars/polars/series/series.py

```diff
@@ -364,6 +368,14 @@ def _from_pyseries(cls, pyseries: PySeries) -> Self:
         series._s = pyseries
         return series
 
+    @staticmethod
+    def _check_inner_array_is_compatible(pl_arr: Array, target: ArrayLike) -> None:
+        inner = pl_arr.inner
+        if isinstance(target, np.ndarray):
```

Comment:
This only checks for integer <-> floating point mismatch, but would still permit casting to a different size integer or floating point value?